src/api/app/controllers/webui/configuration_controller.rb

```diff
@@ -1,106 +1,49 @@
-      redirect_to :action => :connect_instance and return
-    end
+  def interconnect; end
 
-    @project = Project.new(name: project_name)
-    @project.title = params[:title]
-    @project.description = params[:description]
-    @project.remoteurl = params[:remoteurl]
+  def create_interconnect
+    @project = RemoteProject.new(project_params)
 
-    if @project.store
-      flash[:notice] = "Project '#{project_name}' was created successfully"
+    if @project.valid? && @project.store
+      flash[:notice] = "Project '#{@project.name}' was created successfully"
       logger.debug "New remote project with url #{@project.remoteurl}"
-      redirect_to :controller => :project, :action => 'show', :project => project_name and return
+      redirect_to controller: :project, action: 'show', project: @project.name
     else
-      flash.now[:error] = "Failed to save project '#{@project}'"
+      redirect_to :back, error: "Project can't be saved: #{@project.errors.full_messages.to_sentence}"
     end
   end
 
-  def update_configuration
-    if ! (params[:name]  || params[:title] || params[:description])
-      flash[:error] = 'Missing arguments (name, title or description)'
-      redirect_back_or_to :action => 'index' and return
-    end
-
-    begin
-      archs = params[:archs] || []
-      archs.each do |archname, value|
-        available = value == '1'
-        if old = Architecture.where(name: archname, available: !available).first
-          old.available = available
-          old.save
-        end
+  def update
+    respond_to do |format|
+      if @configuration.update(configuration_params)
+        format.html { redirect_to configuration_path, notice: 'Configuration was successfully updated.' }
+      else
+        format.html { redirect_to :back, error: "Configuration can't be saved: #{@configuration.errors.full_messages.to_sentence}" }
       end
-      c = ::Configuration.first
-      c.title = params[:title]
-      c.description = params[:description]
-      c.name = params[:name]
-      c.save
-      flash[:notice] = 'Updated configuration'
-      Rails.cache.delete('configuration')
-    rescue ActiveXML::Transport::Error 
-      logger.debug 'Failed to update configuration'
-      flash[:error] = 'Failed to update configuration'
     end
-    redirect_to :action => 'index'
   end
 
-  def update_architectures
-    @available_architectures.each do |arch_elem|
-      arch = Architecture.find_by_name(arch_elem.name) # fetch a real 'Architecture' from 'directory' entry
-      if params[:arch_available] and params[:arch_available].include?(arch.name) and !arch.available
-        arch.available = true
-        arch.save
-      elsif arch.available
-        arch.available = false
-        arch.save
-      end
-    end
-    redirect_to :action => 'index'
-  end
+  private
 
-  def notifications
-    notifications_for_user(nil)
-  end
+    # Never trust parameters from the scary internet, only allow the white list through.
```

Comment:
We should really do that ...

Here's the rubucop file they came up with in the CSM workshop
https://github.com/SUSE/style-guides/blob/master/rubocop-suse.yml